### ISSUE TITLE
New users going to text editor, confused about what to do

### ISSUE THREAD

**Issue by takluyver** [jupyter/notebook#2182]
See ipython/ipython#10291, but this is also an issue I've seen working with people in person. People have heard that Jupyter is the thing to use, so they install it, open up a `.py` file, and then wonder: How do I run the code? Where are all the buttons I see in screenshots of Jupyter? What's so great about having a text editor in my web browser?

Can we better direct people towards notebooks? Could we find a way that the first thing people see in Jupyter is a notebook interface rather than a file list?

One obvious response is to improve the text editor and better integrate it with our kernel machinery. I think I've seen that this is what Jupyterlab does. But there are other projects expending masses of effort on making text editors and IDEs, and I don't think we're going to make something world-class in that space. Jupyter's central value is notebooks, so even with a better text editor, I think it's worth thinking about how we make it obvious to new users to create a notebook.

**Comment by gnestor** [comment]
It seems to me that this could be accomplished with a dashboard tour, like the [notebook tour](https://github.com/jupyter/notebook/blob/666ecbf35c3310335a3c8c182e8f53441c991c14/notebook/static/notebook/js/tour.js)?

**Comment by rgbkrk** [comment]
Wow that's awesome, I didn't expect that.

I'd really like to make the introductory dashboard more like this to make creating a new notebook more apparent and with less clicks:

![image](https://cloud.githubusercontent.com/assets/836375/22907248/043fd456-f1fd-11e6-99d2-1f5132b95cf8.png)

It's more like a google docs style intro - the documents you can create are at the top. Something that needs to go in the mockup here is text above those document placeholders that reads "Start a new notebook".

**Comment by gnestor** [comment]
The dashboard could use a little redesign. Like lab's launcher and this mockup, the dashboard could display a "New to Jupyter Notebook" at the top that can be dismissed but ideally brought back. I'd also like to see a fixed header so that when the user scrolls down below the fold, he/she can still see actions for selected files, sort directions, and jump to the other tabs without losing scroll position within the file list.

![image](https://cloud.githubusercontent.com/assets/512354/22908051/2b16f510-f201-11e6-9a63-c92d23742c1c.png)

**Comment by takluyver** [comment]
I like the idea of making the 'new notebook' action(s) much more prominent in some form or another. I'd lean towards something like Kyle's mockup, because it emphasises notebooks, which people get Jupyter for, over text editors & terminals. But maybe there could even be different dashboards configurable for different use cases? Text editors & terminals are valuable if you're logging in to Jupyterhub on a server, but I hardly ever use them when I run the notebook locally.

An obvious addition to Kyle's mockup would be for the right-most box to be an 'Install more languages' link leading to the long-awaited kernel directory.

I thought about a tour, but I'd consider that a fallback if we can't think of anything better. If it starts automatically, it's annoying, and if it doesn't, ~no-one will use it. The notebook tour (which doesn't start automatically, to avoid annoying people) was broken for quite a long time before anyone noticed.

**Comment by minrk** [comment]
Is this something we should change on this repo, or focus on JupyterLab doing a better job?

**Comment by takluyver** [comment]
That may be what we ultimately decide, but I've seen this enough times and JupyterLab still feels far enough off that I thought it was worth discussing what we might do in this repo.

**Comment by Carreau** [comment]
What about having a placeholder text in the text editor. And swapping the Notebook section to be on top in the new menu ?

**Comment by Carreau** [comment]
see #2189

**Comment by rgbkrk** [comment]
@carreau I think that generally that little "New" button is eclipsed by the rest that's on the page.

**Comment by Carreau** [comment]
> @carreau I think that generally that little "New" button is eclipsed by the rest that's on the page.

Yeah, but I don't want a full redesign now, that close to 5.0. Swapping seem to be the least annoying/least breaking current user expectations with the more chance to actually have an impact As user "click on the first thing". 

I agree you mockup is much better.

**Comment by takluyver** [comment]
Oh, thanks Matthias. I wasn't actually planning to do anything about this for 5.0, I just wanted to kick off a discussion about it before I forgot. I've bumped it to 5.1 so we can have another look, but I think your PR improves things a bit.
